core/src/Cache/Keys/MsalAccountCacheKey.cs

```diff
@@ -26,30 +26,62 @@
 //------------------------------------------------------------------------------
 
 using System;
-using System.Collections.Generic;
-using System.Linq;
 using System.Text;
-using System.Threading.Tasks;
 
 namespace Microsoft.Identity.Core.Cache
 {
-    internal class MsalAccountCacheKey : MsalCacheKeyBase
+    /// <summary>
+    /// An object representing the key of the token cache Account dictionary. The 
+    /// format of the key is not important for this library, as long as it is unique.
+    /// </summary>
+    internal class MsalAccountCacheKey 
     {
-        public MsalAccountCacheKey(string environment, string tenantId, string userIdentifier) : base(environment, userIdentifier)
+        private readonly string _environment;
+        private readonly string _homeAccountId;
+        private readonly string _tenantId;
+
+        public MsalAccountCacheKey(string environment, string tenantId, string userIdentifier)
         {
-            TenantId = tenantId;
+            if (string.IsNullOrEmpty(environment))
+            {
+                throw new ArgumentNullException(nameof(environment));
+            }
+
+            _tenantId = tenantId;
+            _environment = environment;
+            _homeAccountId = userIdentifier;
         }
-        internal string TenantId { get; set; }
 
         public override string ToString()
         {
             var stringBuilder = new StringBuilder();
 
-            stringBuilder.Append(HomeAccountId + CacheKeyDelimiter);
-            stringBuilder.Append(Environment + CacheKeyDelimiter);
-            stringBuilder.Append(TenantId);
+            stringBuilder.Append(_homeAccountId + MsalCacheConstants.CacheKeyDelimiter);
+            stringBuilder.Append(_environment + MsalCacheConstants.CacheKeyDelimiter);
+            stringBuilder.Append(_tenantId);
 
             return stringBuilder.ToString();
         }
+
+        #region iOS
+
+        public string GetiOSAccountKey()
```

Comment:
Account is also missing generic field

Reply:
It is because it is set to local_account_id which is not part of the account key object.

---
In reply to: [221720840](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1287#discussion_r221720840) [](ancestors = 221720840)

Reply:
Hmm, I think I saw generic in some credential cache keys. Why we would have it differently between accounts and credentials?

Reply:
Yes, you are right, added.

---
In reply to: [222163627](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1287#discussion_r222163627) [](ancestors = 222163627)